### PR TITLE
Prerelease/navigate to record tests

### PR DESCRIPTION
**Pull request by codefriar**
### What does this PR do?

Adds tests to the navigateToRecord LWC. 
As part of the tests, it adds a new tests/lightning/navigation.js stub for the navigation mixin. Pulled from lwc-recipes verbatim.
It also adds a mapping clause to the jest.config.js file.

## The PR fulfills these requirements:

[x] Tests for the proposed changes have been added/updated.
[x] Code linting and formatting was performed.

### Functionality After

<insert gif and/or summary>
image

### CODE REVIEW

File: force-app/main/default/lwc/navigateToRecord/__tests__/navigateToRecord.test.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        element.label = 'foo';
+        document.body.appendChild(element);
+
+        return Promise.resolve().then(() => {
+            // Get handle to anchor and fire click event
+            const anchorEl = element.shadowRoot.querySelector('a');
+            expect(anchorEl).not.toBeNull();
+            expect(anchorEl.textContent).toEqual(element.label);
+            expect.stringMatching(anchorEl.href, /#$/);
+            expect(anchorEl.target).toEqual('_blank');
```
Comment (pozil) [inline force-app/main/default/lwc/navigateToRecord/__tests__/navigateToRecord.test.js:47]: No need to test `href` or `target`.
We only need to test the dynamic part of the template: the label.

File: force-app/main/default/lwc/navigateToRecord/__tests__/navigateToRecord.test.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import { getNavigateCalledWith } from 'lightning/navigation';
+
+describe('c-navigate-to-record', () => {
+    afterEach(() => {
+        // The jsdom instance is shared across test cases in a single file so reset the DOM
+        while (document.body.firstChild) {
+            document.body.removeChild(document.body.firstChild);
+        }
+    });
+    it('navigates to record view by ID', () => {
```
Comment (pozil) [inline force-app/main/default/lwc/navigateToRecord/__tests__/navigateToRecord.test.js:12]: Add a line break.

File: force-app/main/default/lwc/navigateToRecord/__tests__/navigateToRecord.test.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+            const { pageReference } = getNavigateCalledWith();
+
+            expect(pageReference.type).toEqual('standard__recordPage');
+            expect(pageReference.attributes.recordId).toEqual(element.recordId);
+            expect(pageReference.attributes.actionName).toEqual('view');
+        });
+    });
+
+    it('creates an anchor element with a blank target, # href and label', () => {
```
Comment (pozil) [inline force-app/main/default/lwc/navigateToRecord/__tests__/navigateToRecord.test.js:34]: Update to: "creates an anchor element with the right label"

**Comment by codefriar** [comment]
@pozil all comments resolved, please reelaluate

**Pull request merged**